pkg/networkconfigmanagement/sender/sender.go

```diff
@@ -11,31 +11,76 @@ package sender
 import (
 	"encoding/json"
 	"fmt"
+	"time"
+
 	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
 	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/networkconfigmanagement/profile"
 	ncmreport "github.com/DataDog/datadog-agent/pkg/networkconfigmanagement/report"
+	"github.com/DataDog/datadog-agent/pkg/networkdevice/utils"
+	"github.com/benbjohnson/clock"
+)
+
+const (
+	ncmCheckDurationMetric  = "datadog.ncm.check_duration"
+	ncmCheckIntervalMetric  = "datadog.ncm.check_interval"
+	ncmConfigSizeMetric     = "ncm.config_size"
+	ncmRunningConfigTypeTag = "config_type:running"
+	ncmStartupConfigTypeTag = "config_type:startup"
 )
 
 // NCMSender is a wrapper around the sender.Sender to send network device configuration data
 type NCMSender struct {
-	Sender    sender.Sender
-	namespace string
+	Sender     sender.Sender
+	namespace  string
+	hostname   string // TODO: get the hostname to use
+	deviceTags []string
+	clock      clock.Clock
 }
 
 // NewNCMSender creates a new NCMSender
-func NewNCMSender(sender sender.Sender, namespace string) *NCMSender {
+func NewNCMSender(sender sender.Sender, namespace string, clock clock.Clock) *NCMSender {
 	return &NCMSender{
 		Sender:    sender,
 		namespace: namespace,
+		clock:     clock,
 	}
 }
 
-// SendNCMMetrics sends the network device configuration metrics to Datadog
-func (s *NCMSender) SendNCMMetrics() error {
-	// Implement the logic to send the network device configuration
-	// This is a placeholder implementation
-	//s.GaugeWithTimestamp()
-	return nil
+// SetDeviceTags sets the device tags for the sender to use for metrics submission
+func (s *NCMSender) SetDeviceTags(deviceTags []string) {
+	s.deviceTags = deviceTags
+}
+
+func (s *NCMSender) getDeviceTags() []string {
+	return utils.CopyStrings(s.deviceTags)
+}
+
+// SendNCMCheckMetrics sends metrics about the check itself to Datadog
+func (s *NCMSender) SendNCMCheckMetrics(startTime time.Time, lastCheckTime time.Time) {
+	tags := append(s.getDeviceTags(), utils.GetCommonAgentTags()...)
+	duration := s.clock.Since(startTime).Seconds()
+	s.Sender.Gauge(ncmCheckDurationMetric, duration, s.hostname, tags)
+
+	if !lastCheckTime.IsZero() {
+		interval := startTime.Sub(lastCheckTime).Seconds()
+		s.Sender.Gauge(ncmCheckIntervalMetric, interval, s.hostname, tags)
+	}
+}
+
+// SendMetricsFromExtractedMetadata sends metrics from data extracted from the device config after processing
+func (s *NCMSender) SendMetricsFromExtractedMetadata(metadata profile.ExtractedMetadata, configType ncmreport.ConfigType) {
+	tags := s.getDeviceTags()
```

Comment:
should tags be the same across all metrics ? (in case we need to cross both for troubleshooting ?)
```suggestion
	tags := append(s.getDeviceTags(), utils.GetCommonAgentTags()...)
```